### PR TITLE
add patch & revoke permissions

### PR DESCRIPTION
**Pull request by aenario**

### CODE REVIEW

**Comment by nono** [comment]
I didn't read the PR yet. Just two things:

1. There is a conflit, the PR needs to be rebased
2. I don't see documentation

**Comment by aenario** [comment]
1. Fixed
2. You already wrote most of it (this implements the APIs defined in docs/permissions.md)

File: docs/permissions.md [hunk 1]
```diff
@@ -369,6 +369,32 @@ Accept: application/vnd.api+json
...
+```json
+{
+  "data": {
+    "id": "a340d5e0-d647-11e6-b66c-5fc9ce1e17c6",
+    "type": "io.cozy.permissions",
+    "permissions": {
+      "add-this": {
+        "type": "io.cozy.files",
+        "verbs": ["GET"],
+        "values": ["some-picture-id"],
```
Comment (nono) [inline docs/permissions.md]: [Nitpick] no dangling comma here (JSON)

File: web/permissions/permissions.go [hunk 2]
```diff
@@ -129,10 +138,86 @@ func listPermissions(c echo.Context) error {
...
+	if err != nil {
+		return err
+	}
+
+	var patch permissions.Permission
+	if _, err = jsonapi.Bind(c.Request(), &patch); err != nil {
+		return err
+	}
+
+	patchSet := patch.Permissions != nil && len(*patch.Permissions) > 0
```
Comment (nono) [inline web/permissions/permissions.go]: Just a question: why `patch.Permissions` is a pointer to a slice, and not just a slice? It's not common to need a pointer to a slice, as a slice is already a kind of pointer.

Comment (aenario) [inline web/permissions/permissions.go]: Permissions is not a slice, but a permissions.Set. When testing custom marshalling/unmarshalling of this field (as a JSON.object[rule tile] -> rule), i thought I needed it to be a pointer, tourns out I was wrong, fix below.

File: docs/permissions.md (hunk 1 above)
Comment (aenario) [inline docs/permissions.md]: fixed

**Pull request merged**